core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -314,7 +314,54 @@ struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
   }
   bool impl_tune_tile_size() const { return m_tune_tile_size; }
 
+  tile_type tile_size_recommended() const {
+    tile_type rec_tile_sizes = {};
+
+    for (std::size_t i = 0; i < rec_tile_sizes.size(); ++i) {
+      rec_tile_sizes[i] = tile_size_recommended(i);
+    }
+    return rec_tile_sizes;
+  }
+
+  int tile_size_recommended(const int tile_rank) const {
+    auto properties   = Impl::get_tile_size_properties(m_space);
+    int last_rank     = (inner_direction == Iterate::Right) ? rank - 1 : 0;
+    int rec_tile_size = properties.default_tile_size;
+
+    if (tile_rank == last_rank) {
+      rec_tile_size = tile_size_last_rank(
+          properties, m_upper[last_rank] - m_lower[last_rank]);
+    }
+    return rec_tile_size;
+  }
+
+  tile_type largest_tile_size_recommended() const {
+    tile_type rec_largest_tile_sizes = {};
+
+    for (std::size_t i = 0; i < rec_largest_tile_sizes.size(); ++i) {
+      rec_largest_tile_sizes[i] = largest_tile_size_recommended(i);
+    }
+    return rec_largest_tile_sizes;
+  }
+
+  int largest_tile_size_recommended(const int tile_rank) const {
+    auto properties = Impl::get_tile_size_properties(m_space);
+    return tile_size_last_rank(properties,
+                               m_upper[tile_rank] - m_lower[tile_rank]);
+  }
```

Comment:
Lets only expose `tile_size_recommended` and `max_total_tile_size`, I am not sure how a user would correctly use `largest_tile_size_recommended` which really is more like a "recommended tile size in the rank which corresponds to the fastest iteration".

Reply:
Removed `largest_tile_size_recommended()`.
I have no strong opinion on keeping or removing it. But for Cuda/HIP/SYCL, the tile size of '16' is used by default in the rank with the fastest iteration. Users won't have access to this default number without this function.